static/app/components/globalDrawer/components.tsx

```diff
@@ -28,30 +41,175 @@ interface DrawerPanelProps {
   children: React.ReactNode;
   headerContent: React.ReactNode;
   onClose: DrawerContentContextType['onClose'];
+  drawerKey?: string;
   drawerWidth?: DrawerOptions['drawerWidth'];
   transitionProps?: AnimationProps['transition'];
 }
 
+function getDrawerWidthKey(drawerKey: string) {
+  return `drawer-width:${drawerKey}`;
+}
+
 export function DrawerPanel({
   ref,
   ariaLabel,
   children,
   transitionProps,
   onClose,
   drawerWidth,
+  drawerKey,
 }: DrawerPanelProps & {
   ref?: React.Ref<HTMLDivElement>;
 }) {
+  const resizeHandleRef = useRef<HTMLDivElement>(null);
+  const panelRef = useRef<HTMLDivElement>(null);
+  const rafIdRef = useRef<number | null>(null);
+  const initialMousePositionRef = useRef<number | null>(null);
+
+  // Calculate initial width from props or use default
+  const calculateInitialWidth = (savedWidth?: number) => {
+    // If we have a saved width, use it but ensure it's within bounds
+    if (savedWidth !== undefined) {
+      return Math.min(Math.max(savedWidth, MIN_WIDTH_PERCENT), MAX_WIDTH_PERCENT);
+    }
+
+    if (drawerWidth) {
+      // If width is already in percentage, parse and clamp it
+      if (drawerWidth.endsWith('%')) {
+        const parsedPercent = parseFloat(drawerWidth);
+        return Math.min(Math.max(parsedPercent, MIN_WIDTH_PERCENT), MAX_WIDTH_PERCENT);
+      }
+      // If width is in pixels, convert to percentage and clamp
+      const viewportWidth = window.innerWidth;
+      const parsedPixels = parseFloat(drawerWidth);
+      const percentValue = (parsedPixels / viewportWidth) * 100;
+      return Math.min(Math.max(percentValue, MIN_WIDTH_PERCENT), MAX_WIDTH_PERCENT);
+    }
+
+    return DEFAULT_WIDTH_PERCENT;
+  };
+
+  // Store persisted width in localStorage, but don't use it for rendering state
+  const [persistedWidthPercent, setPersistedWidthPercent] =
+    useSyncedLocalStorageState<number>(
+      drawerKey ? getDrawerWidthKey(drawerKey) : 'drawer-width:default',
+      (value?: unknown) => {
+        const savedWidth = typeof value === 'number' ? value : undefined;
+        return calculateInitialWidth(savedWidth);
+      }
+    );
+
+  useLayoutEffect(() => {
+    if (panelRef.current) {
+      panelRef.current.style.setProperty('--drawer-width', `${persistedWidthPercent}%`);
+      panelRef.current.style.setProperty('--drawer-min-width', `${MIN_WIDTH_PERCENT}%`);
+      panelRef.current.style.setProperty('--drawer-max-width', `${MAX_WIDTH_PERCENT}%`);
```

Comment:
Why do we have to set these here if we're already setting them in the styled component below, they're constants right?